src/main/kotlin/it/unibo/protelis/web/execution/simulated/NewSimulatedProtelisEngine.kt

```diff
@@ -0,0 +1,82 @@
+package it.unibo.protelis.web.execution.simulated
+
+import it.unibo.alchemist.core.implementations.Engine
+import it.unibo.alchemist.core.interfaces.Simulation
+import it.unibo.alchemist.core.interfaces.Status
+import it.unibo.alchemist.core.interfaces.waitForAndCheck
+import it.unibo.alchemist.loader.YamlLoader
+import it.unibo.alchemist.model.implementations.positions.Euclidean2DPosition
+import it.unibo.alchemist.model.implementations.times.DoubleTime
+import it.unibo.protelis.web.execution.ProtelisObserver
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class NewSimulatedProtelisEngine {
+  private val logger: Logger = LoggerFactory.getLogger(this::class.java)
+  private var alchemistEngine: Simulation<Any, Euclidean2DPosition>? = null
+
+  fun setup(sourceCode: String, monitor: ProtelisObserver) {
+    if (alchemistEngine != null) {
+      stop()
+    }
+
+    alchemistEngine = setupSimulation()
+    setCode(sourceCode)
+    alchemistEngine
+      ?.addOutputMonitor(ProtelisUpdateAdapter(monitor))
+      ?: throw IllegalStateException("Simulation was not set up correctly")
+    alchemistEngine
+      ?.waitForAndCheck(Status.INIT)
+      ?: throw IllegalStateException("Simulation was not set up correctly")
+    GlobalScope.launch() { alchemistEngine?.run() ?: throw IllegalStateException() }
+    logger.debug("Simulation Engine set up correctly")
+  }
+
+  fun start() {
+    checkNotNull(alchemistEngine)
+    alchemistEngine
+      ?.play()
+      ?: throw IllegalStateException()
+    alchemistEngine
+      ?.waitForAndCheck(Status.RUNNING)
+      ?: throw IllegalStateException()
+    logger.debug("Simulation started")
+  }
+
+  fun stop() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [UndocumentedPublicFunction - [stop] at /src/src/main/kotlin/it/unibo/protelis/web/execution/simulated/NewSimulatedProtelisEngine.kt:48:3](https://app.codacy.com/manual/NiccoMlt/Protelis-Web/pullRequest?prid=5040950)